code/modules/urist/modules/weather/weather_defines.dm

```diff
@@ -0,0 +1,168 @@
+#define WEATHER_ACTION_VOLUME 10 //how much of the reagents transfers per act
+//Those are objs only because they need to have access to some vars, *DO NOT* USE THEM DIRECTLY!
+/obj/weathertype
+	name = "DO NOT USE THIS, THIS IS ONLY AN OVERLAY!"
+	icon = 'icons/urist/weather.dmi'
+	icon_state = ""
+	color = null
+	alpha = 255
+	mouse_opacity = 0
+	anchored = 1
+	layer = 5
+	var/weathertemp = 310.15 //Kelvin temperature, default is neutral to mobs
+	var/list/init_reagents = list() //carried reagents as ids
+
+/obj/weathertype/New()
+	. = ..()
+	if(init_reagents.len)
+		create_reagents(999)
+		for(var/R in init_reagents) //TODO: secure this from non-reagent ID strings
+			reagents.add_reagent(R, 50, null, 1)
+
+/obj/weathertype/proc/GetWeatherEffect(var/turf/T) //handles specific effects of weather on... stuff
+	return 1
+
+/*hijacked from human/life.dm's handle_environment and simplified;
+all-purpose cold/hot weather helper for exposure effects, wear a hat */
+/obj/weathertype/proc/TemperatureEffect(var/mob/living/M)
+	//Body temperature adjusts depending on surrounding atmosphere based on your thermal protection (convection)
+	var/temp_adj = 0
+	if(weathertemp < M.bodytemperature)			//Place is colder than we are
```

Comment:
So the weather doesn't actually change the temperature of the turf? That's good, the less atmos overhead, the better.